src/kernels/jupyter/session/jupyterSession.ts

```diff
@@ -180,12 +174,19 @@ export class JupyterSession extends BaseJupyterSession {
     }
 
     protected startRestartSession(disableUI: boolean) {
-        if (!this.restartSessionPromise && this.session && this.contentsManager) {
-            const token = new CancellationTokenSource();
-            const promise = this.createRestartSession(disableUI, this.session, token.token);
-            this.restartSessionPromise = { token, promise };
-            promise.finally(() => token.dispose());
+        if (!this.session) {
+            throw new Error('Session disposed or not initialized');
```

Comment:
its a shame we still need this. because `this.session` can be null, hopefully it will go away in a subsequent refactor